File: FizzBuzz/TDD/kotlin/src/main/kotlin/PrintersProvider.kt
object PrintersProvider {

	private val printers: MutableList<(List<String>) -> String> = mutableListOf()

	fun add(printer: (List<String>) -> String)
	{
		printers.add(printer)
	}

	fun removeAll()
	{
		printers.clear()
	}

	fun all(): MutableList<(List<String>) -> String>
	{
		return printers.toMutableList()
	}
}

File: FizzBuzz/TDD/Scala/src/main/scala/PrintersProvider.scala
object PrintersProvider {
  private var printers : List[(Any) => Any] = Nil

  def all(): List[(Any) => Any] =
  {
    printers
  }

  def removeAll()
  {
    printers = Nil
  }

  def add(printer: (Any) => Any)
  {
    printers = printer::printers
  }
}
